phylogenetic/rules/merge_additional_inputs.smk

```diff
@@ -0,0 +1,152 @@
+"""
+This part of the workflow merges additional metadata and sequences.
+
+REQUIRED INPUTS:
+
+    config = "defaults/config.yaml" with an `input` dictionary of metadata and sequence pairs
+
+OUTPUTS:
+
+    merged_sequences = results/sequences_merged.fasta
+    merged_metadata = results/metadata_merged.tsv
+
+This part of the workflow usually includes the following steps:
+
+    - augur merge
+See Augur's usage docs for these commands for more details.
+"""
+
+# ------------- helper functions to collect, merge & download input files ------------------- #
+NEXTSTRAIN_PUBLIC_BUCKET = "s3://nextstrain-data/"
+
+def _parse_config_input(input):
+    """
+    Parses information from an individual config-defined input, i.e. an element within `config.inputs` or `config.additional_inputs`
+    and returns information snakemake rules can use to obtain the underlying data.
+
+    The structure of `input` is a dictionary with keys:
+    - name:string (required)
+    - metadata:string (optional) - a s3 URI or a local file path
+    - sequences:string|dict[string,string] (optional) - either a s3 URI or a local file path, in which case
+      it must include a '{segment}' wildcard substring, or a dict of segment → s3 URI or local file path,
+      in which case it must not include the wildcard substring.
+
+    Returns a dictionary with optional keys:
+    - metadata:string - the relative path to the metadata file. If the original data was remote then this represents
+      the output of a rule which downloads the file
+    - metadata_location:string - the URI for the remote file if applicable else `None`
+    - sequences:function. Takes in wildcards and returns the relative path to the sequences FASTA for the provided
+      segment wildcard, or returns `None` if this input doesn't define sequences for the provided segment.
+    - sequences_location:function. Takes in wildcards and returns the URI for the remote file, or `None`, where applicable.
+
+    Raises InvalidConfigError
+    """
+    name = input['name']
+    lambda_none = lambda w: None
+
+    info = {'metadata': None, 'metadata_location': None, 'sequences': lambda_none, 'sequences_location': lambda_none}
+
+    def _source(uri, *,  s3, local):
+        if uri.startswith('s3://'):
+            return s3
+        elif uri.lower().startswith('http://') or uri.lower().startswith('https://'):
+            raise InvalidConfigError("Workflow cannot yet handle HTTP[S] inputs")
```

Comment:
Do we want to add support for HTTP[S] inputs?

Reply:
Yes, but there may be better ways of doing it than this. See [this reply to the zika PR](https://github.com/nextstrain/zika/pull/80#pullrequestreview-2638275736).

Reply:
Thank you! I'm comfortable continuing to use S3 inputs for WNV for now, as it simplifies documenting the API